homeassistant/components/media_player/channels.py

```diff
@@ -73,22 +69,23 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
 
     def service_handler(service):
         """Handler for services."""
-        entity_ids = service.data.get(ATTR_ENTITY_ID)
-
-        if entity_ids:
-            devices = [device for device in hass.data[DATA_CHANNELS]
-                       if device.entity_id in entity_ids]
-        else:
-            devices = hass.data[DATA_CHANNELS]
-
-        for device in devices:
-            if service.service == SERVICE_SEEK_FORWARD:
-                device.seek_forward()
-            elif service.service == SERVICE_SEEK_BACKWARD:
-                device.seek_backward()
-            elif service.service == SERVICE_SEEK_BY:
-                seconds = service.data.get('seconds')
-                device.seek_by(seconds)
+        entity_id = service.data.get(ATTR_ENTITY_ID)
+
+        device = next([device for device in hass.data[DATA_CHANNELS] if
```

Comment:
Use parenthesis instead of brackets and drop the `__iter__`, to have it be a generator expression.